python/strings_udf/strings_udf/__init__.py

```diff
@@ -59,13 +64,34 @@ def compiler_from_ptx_file(path):
                 "files. Please file an issue reporting this error and how you "
                 "installed cudf and strings_udf."
             )
-        sms = [
-            os.path.basename(f).rstrip(".ptx").lstrip("shim_") for f in files
-        ]
-        selected_sm = max(sm for sm in sms if sm < cc)
-        ptxpath = os.path.join(
-            os.path.dirname(__file__), f"shim_{selected_sm}.ptx"
-        )
+
+        suffix_a_sm = None
+        regular_sms = []
+
+        for f in files:
+            file_name = os.path.basename(f)
+            sm_number = file_name.rstrip(".ptx").lstrip("shim_")
+            if sm_number.endswith("a"):
+                processed_sm_number = int(sm_number.rstrip("a"))
+                if processed_sm_number == cc:
+                    suffix_a_sm = (processed_sm_number, f)
+            else:
+                regular_sms.append((int(sm_number), f))
+
+        regular_result = None
+
+        if regular_sms:
+            regular_result = _get_appropriate_file(regular_sms, cc)
+
+        if suffix_a_sm is None and regular_result is None:
+            raise RuntimeError(
+                "This strings_udf installation is missing the necessary PTX "
+                f"files that are <={cc}."
+            )
+        elif suffix_a_sm is not None:
+            ptxpath = suffix_a_sms[0][1]
```

Comment:
```suggestion
            ptxpath = suffix_a_sms[1]
```

Right or we are taking the first character out of the string?

Reply:
:)